### ISSUE TITLE
tdspool and 20017 Unexpected EOF from the server

### ISSUE THREAD

**Issue by vstaf** [FreeTDS/freetds#70]
Hello @freddy77 !
We are mirgating to tdspool on production. Everything works fine and thank you for you great work!
But today i got strange errors 
SQLSTATE[01002]: Disconnect error: 20017 Unexpected EOF from the server [20017] (severity 9) [(null)]
when try to execute query (testing with insert and update) with huge amount of text in one field (~100k symbols).
In sql server logs (2014 sql server) i see:

Date		01.07.2016 16:39:36
Log		SQL Server (Current - 30.06.2016 17:59:00)
Source		spid180
Message
Error: 4014, Severity: 20, State: 17.

Date		01.07.2016 16:39:36
Log		SQL Server (Current - 30.06.2016 17:59:00)
Source		spid180
Message
A fatal error occurred while reading the input stream from the network. The session will be terminated (input error: 0, output error: 0).

Over queries executed successfuly in this time. I restarted tdspool and problem was gone.
Do you have any ideas what's the problem?

I try to reproduce this error now but cann't>.< I think the problem appear after some hours / days tdspool work. I enabled dump and wating now.

It's only one problem that we got after migrating to tdspool on production. It works stable without any crashes. We are using it with centos 6, php-fpm 5.5 and 2014 sql server.

**Comment by freddy77** [comment]
I have no idea. tdspool should not change the stream and the fact that without tdspool works seems to indicate that the stream is usually correct.

**Comment by vstaf** [comment]
Ok, i see

**Comment by vstaf** [comment]
@freddy77 hello again!
I can 100% reproduce this problem now, but couldn't collect any debug information because when i enable dump file directive in freetds.conf the problem disappear and back again after i comment #dump file. Could you tell me any another way to collect some usefull info for you?

```
freetds.conf

[global]
        tds version = 7.4
#       dump file = /var/log/freetds.log

[mssql]
host = x.x.x.x
port = 1433
client charset = UTF-8


pool.conf

[global]
        min pool conn = 10
        max pool conn = 100
        max member age = 0

[social]
        min pool conn = 50
        max pool conn = 100
        max member age = 0
        server = mssql
        user = php
        password = 
        database = db
        port = 5000
```

Sample script for reproduce this issue doing 1 query, where plainText and htmlText parameters contain sample large piece of text (25000 col, don't know what size is "critical")

query like this:
```
INSERT INTO tmp.table (
                docType
                , userId
                , serviceId
                , language
                , reason
                , plainText
                , htmlText
                , active
                , createDate
            )
            OUTPUT INSERTED.id
            VALUES (
                ?
                , ?
                , ?
                , ?
                , ?
                , ?
                , ?
                , 0
                , getdate()
            )';
```
When i try run script i've got 
SQLSTATE[01002]: Disconnect error: 20017 Unexpected EOF from the server [20017] (severity 9) [(null)]
error in php logs, query didn't insert any data to the table.
In SQL server logs i see Error: 4014, Severity: 20, State: 17. I googled this error and try to disable TCP/IP Chimney Offload and RSS on the server but it didn't help.

**Comment by freddy77** [comment]
Great!
I would do:
- try to get a TDSDUMP not using the pool;
- try using TDSDUMP of just the single client with the pool;
- use a network capture (as wireshark)
As usual post to me if they are too big or worried about sensitive data (the suggestion is always to use test environment).

**Comment by vstaf** [comment]
Here is tcpdump output. Lauching pool (with settings that i posted before) and execute script.
[tdspool.zip](https://github.com/FreeTDS/freetds/files/376112/tdspool.zip)

**Comment by freddy77** [comment]
Found something strange.
Can you confirm that same program works directly connected to the server.
Actually, can you post a tcpdump of that?

**Comment by freddy77** [comment]
Actually looks like a big TDS packet (4096 bytes) is split in 3 TCP (MTU = 1500 so make sense) but last packet is not transmitted :(

**Comment by freddy77** [comment]
Got a repro (had to modify the code, weird).
Got a fix, I'll check some more stuff.

**Comment by vstaf** [comment]
But i've no idea why everything works correctly when i enable dump file directive in freetds.conf. Here you are tcpdump.cap with packets where script insert data in the table successfuly.

btw, i'll try to reproduce this issue without using 
[tdspool_good.zip](https://github.com/FreeTDS/freetds/files/377707/tdspool_good.zip)
tdspool today and come back with results.

**Comment by vstaf** [comment]
wow, it's good news, ty!)

**Comment by freddy77** [comment]
Fixed in master and Branch-1.0.
New snapshot tomorrow.
